aclk/aclk.c

```diff
@@ -895,3 +972,92 @@ struct label *add_aclk_host_labels(struct label *label) {
     return label;
 #endif
 }
+
+void aclk_host_state_update(RRDHOST *host, int cmd) {
+    uuid_t node_id;
+    int ret;
+
+    if (!aclk_connected || !aclk_use_new_cloud_arch)
+        return;
+
+    ret = get_node_id(&host->host_uuid, &node_id);
+    if (ret > 0) {
+        // this means we were not able to check if node_id already present
+        error("Unable to check for node_id. Ignoring the host state update.");
+        return;
+    }
+    if (ret < 0) {
+        // node_id not found
+        aclk_query_t create_query;
+        create_query = aclk_query_new(REGISTER_NODE);
+        rrdhost_aclk_state_lock(localhost);
+        create_query->data.node_creation.claim_id = strdupz(localhost->aclk_state.claimed_id);
+        rrdhost_aclk_state_unlock(localhost);
+        create_query->data.node_creation.hops = 1; //TODO
+        create_query->data.node_creation.hostname = strdupz(host->hostname);
+        create_query->data.node_creation.machine_guid = strdupz(host->machine_guid);
+        aclk_queue_query(create_query);
+        return;
+    }
+
+    aclk_query_t query = aclk_query_new(NODE_STATE_UPDATE);
+    query->data.node_update.hops = 1; //TODO
```

Comment:
Please, can you add more information for what is necessary `TODO`.

Reply:
added clarification - this will be changed when streaming hops will be implemented (e.g. when we will have information from streaming how far the children is from parent)